mesonbuild/coredata.py

```diff
@@ -776,8 +776,7 @@ def add_lang_args(self, lang: str, comp: T.Type['Compiler'],
                       for_machine: MachineChoice, env: 'Environment') -> None:
         """Add global language arguments that are needed before compiler/linker detection."""
         from .compilers import compilers
-        options = compilers.get_global_options(lang, comp, for_machine, env)
-        self.add_compiler_options(options, lang, for_machine, env)
+        self.options.update(compilers.get_global_options(lang, comp, for_machine, env))
```

Comment:
Are you sure it should use update() here and not setdefault()? Is add_lang_args() is called only once per language so we are sure those options does not exist yet?

Reply:
This is when the language options are created for the first time, and it's called once per language, per machine.